ssc/Pos/Binary/Ssc/Core.hs

```diff
@@ -4,59 +4,62 @@ module Pos.Binary.Ssc.Core
+                                         deriveSimpleBiCxt, encodeListLen, enforceSize)
 import           Pos.Binary.Crypto      ()
 import           Pos.Core.Configuration (HasConfiguration)
 import           Pos.Core.Vss           (VssCertificate (..), VssCertificatesMap (..),
-                                         mkVssCertificatesMap, recreateVssCertificate)
+                                         getCertId, mkVssCertificatesMap,
+                                         recreateVssCertificate)
 import           Pos.Crypto             (Hash, PublicKey)
 import           Pos.Ssc.Core.Types     (Commitment (..), CommitmentsMap (..),
                                          Opening (..), OpeningsMap, SharesMap,
                                          SignedCommitment, SscPayload (..), SscProof (..),
                                          VssCertificatesHash, mkCommitmentsMap)
-import           Serokell.Util          (allDistinct)
 
 instance Bi Commitment where
-  encode Commitment{..} = encodeListLen 2 <> encode commShares
-                                          <> encode commProof
-  decode = do
-    enforceSize "Commitment" 2
-    commShares <- decode
-    when (null commShares) $ fail "decode@Commitment: no shares"
-    commProof <- decode
-    return $ Commitment commProof commShares
+    encode Commitment{..} = encodeListLen 2 <> encode commShares
+                                            <> encode commProof
+    decode = do
+        enforceSize "Commitment" 2
+        commShares <- decode
+        when (null commShares) $ fail "decode@Commitment: no shares"
+        commProof <- decode
+        return $ Commitment commProof commShares
 
 instance Bi CommitmentsMap where
-  encode = encodeCommitments
-  decode = decodeCommitments
+    encode = encodeCommitments
+    decode = decodeCommitments
 
 instance HasConfiguration => Bi VssCertificate where
-  encode vssCert = encodeListLen 4 <> encode (vcVssKey vssCert)
-                                   <> encode (vcExpiryEpoch vssCert)
-                                   <> encode (vcSignature vssCert)
-                                   <> encode (vcSigningKey vssCert)
-  decode = do
-    enforceSize "VssCertificate" 4
-    key <- decode
-    epo <- decode
-    sig <- decode
-    sky <- decode
-    case recreateVssCertificate key epo sig sky of
-      Left e  -> fail e
-      Right v -> pure v
+    encode vssCert = encodeListLen 4 <> encode (vcVssKey vssCert)
+                                     <> encode (vcExpiryEpoch vssCert)
+                                     <> encode (vcSignature vssCert)
+                                     <> encode (vcSigningKey vssCert)
+    decode = do
+        enforceSize "VssCertificate" 4
+        key <- decode
+        epo <- decode
+        sig <- decode
+        pk <- decode
+        ifM (view dcNoCheck)
+            (pure $ VssCertificate key epo sig pk)
+            (recreateVssCertificate key epo sig pk)
```

Comment:
I went to see what `recreateVssCertificate` did in `core/Pos/Core/Vss`, and the Haddock comment says "This function main 'fail' if data is invalid."

Mind changing that to "This function may fail"?